compiler/qsc_qasm3/src/semantic/lowerer.rs

```diff
@@ -1072,8 +1361,8 @@ impl Lowerer {
         // The condition for the switch statement must be an integer type
         // so we use `cast_expr_to_type`, forcing the type to be an integer
         // type with implicit casts if necessary.
-        let target_ty = Type::Int(None, false);
-        let target = self.cast_expr_to_type(&target_ty, &target, target.span);
+        let target_ty = Type::Int(None, target.ty.is_const());
+        let target = self.cast_expr_to_type(&target_ty, &target);
 
         // It is a parse error to have a switch statement with no cases,
         // even if the default block is present. Getting here means the
```

Comment:
This assert needs to be removed since we now try to lower switch stmts with parse errors